src/main/java/frc/robot/subsystems/Dashboard.java

```diff
@@ -107,6 +109,23 @@ private Dashboard()
     
         _autonomousLog = autonomousOptions.add("Auto Log", "").withPosition(0, 4).withSize(1, 1).withWidget(BuiltInWidgets.kTextView).getEntry();
 
+        //Preferences
+        Preferences.getDouble("FL", 90.0);
+        Preferences.getDouble("FR", 90.0);
+        Preferences.getDouble("BL", 90.0);
+        Preferences.getDouble("BR", 90.0);
+
+        Preferences.getDouble("Max Extend", 36.0);
+        Preferences.getDouble("Arm Min", -90.0);
+        Preferences.getDouble("Arm Max", 90.0);
+
+        Preferences.getDouble("Wrist Min", -90.0);
+        Preferences.getDouble("Wrist Max", 90.0);
+        Preferences.getDouble("Twist Min", -180.0);
+        Preferences.getDouble("Twist Max", 180.0);
+        Preferences.getDouble("Eject Time", 1.0);
+        Preferences.getDouble("Intake Speed", 0.75);
+
         var settingsTab                 = Shuffleboard.getTab("Settings");
 
         //Drive
```

Comment:
GitHub won't let me comment on the exact line because there's not an existing change in the relevant area, but down a little way from here around line 140 is where we set up the button for re-zeroing swerve module rotations. The code looks like this:

```java
        var resetSwerveOffsetButton     = new DashboardButton(resetSwerveOffsetEntry);
        resetSwerveOffsetButton.whenPressed(() -> {
            Drive.getInstance().zeroModuleRotations();
            flOffset.setDouble(Drive.getInstance().getSwerveModule(Constants.Drive.FL_INDEX).getRelativeZero());
            frOffset.setDouble(Drive.getInstance().getSwerveModule(Constants.Drive.FR_INDEX).getRelativeZero());
            blOffset.setDouble(Drive.getInstance().getSwerveModule(Constants.Drive.BL_INDEX).getRelativeZero());
            brOffset.setDouble(Drive.getInstance().getSwerveModule(Constants.Drive.BR_INDEX).getRelativeZero());
        });
``` 

The lambda function we create during the `resetSwerveOffsetButton.whenPressed` method call needs to be updated to also update the Preferences for the swerve module offsets. I'm thinking something like this:

```java
        var resetSwerveOffsetButton     = new DashboardButton(resetSwerveOffsetEntry);
        resetSwerveOffsetButton.whenPressed(() -> {
            Drive.getInstance().zeroModuleRotations();
            flOffset.setDouble(Drive.getInstance().getSwerveModule(Constants.Drive.FL_INDEX).getRelativeZero());
            frOffset.setDouble(Drive.getInstance().getSwerveModule(Constants.Drive.FR_INDEX).getRelativeZero());
            blOffset.setDouble(Drive.getInstance().getSwerveModule(Constants.Drive.BL_INDEX).getRelativeZero());
            brOffset.setDouble(Drive.getInstance().getSwerveModule(Constants.Drive.BR_INDEX).getRelativeZero());

            Preferences.setDouble("FL Offset", Drive.getInstance().getSwerveModule(Constants.Drive.FL_INDEX).getRelativeZero());
            Preferences.setDouble("FR Offset", Drive.getInstance().getSwerveModule(Constants.Drive.FR_INDEX).getRelativeZero());
            Preferences.setDouble("BL Offset", Drive.getInstance().getSwerveModule(Constants.Drive.BL_INDEX).getRelativeZero());
            Preferences.setDouble("BR Offset", Drive.getInstance().getSwerveModule(Constants.Drive.BR_INDEX).getRelativeZero());
        });
```